dynamic_recursion_fibonacci.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+* @file
+* @brief [Fibonacci Sequence ](https://en.wikipedia.org/wiki/Generalizations_of_Fibonacci_numbers#Extension_to_negative_integers)
+*
+* @details
+* This program generates both negative and positive fibonacci sequence
+* using recursion and dynamic programming
+* @author [Arunabh Bhattacharya](https://github.com/arunabh-cloud)
+*/
+
+#include <cmath>   /// for absolute value and pow
+#include <vector>  /// for std::vector
+#include <cassert>  /// for assert
+
+/**
+ * @brief Algorithm implementation for fibonacci using recursion
+ * @param n is the input for generating the fibonacci sequence
+ * @returns a negative sequence if n is negative
+ * @returns 0 or 1 if n is 0 or 1
+ * @returns a positive sequence if n is positive
+ */
+int fibRec(int n) {
+    if (n < 0) {
+        return (fibRec(-n)) * (int)pow(-1, abs(n));
+    }
+    else if (n == 0 || n == 1) {
+        return n;
+    }
+    else {
+        return fibRec(n - 2) + fibRec(n - 1);
+    }
+}
+
+/**
+ * brief Algorithm implementation for fibonacci using dynamic programming
+ * @param n is the input for generating the fibonacci sequence
+ * @returns a negative sequence if n is negative
+ * @returns 0 or 1 if n is 0 or 1
+ * @returns a positive sequence if n is positive
+ */
+int fibTab(int n) {
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

Reply:
The function returns both positive and negative values. What should I use for that?